src/js/components/common/minard-link.tsx

```diff
@@ -3,46 +3,56 @@ import { Link } from 'react-router';
 
 import { Branch } from '../../modules/branches';
 import { Comment } from '../../modules/comments';
-import { Commit } from '../../modules/commits';
 import { Deployment } from '../../modules/deployments';
 import { Project } from '../../modules/projects';
 
 interface Props {
-  deployment?: Deployment;
-  preview?: Deployment;
-  commit?: Commit;
-  branch?: Branch | string;
-  project?: Project | string;
-  comment?: Comment;
-  openInNewWindow?: boolean;
+  rawDeployment?: {
+    deployment: Deployment;
```

Comment:
The main purpose is to make it explicit what type of entity you want a link for, and to make it consistent across entity types so that all the props have a similar shape (i.e. an object with properties that contain the actual content).